imap_processing/cdf/tests/test_cdf_attribute_manager.py

```diff
@@ -1,58 +1,322 @@
+    )
+    assert (
+        cdf_manager.variable_attribute_schema["attribute_key"]["RESOLUTION"]["required"]
+        is True
+    )
+
+
 # @pytest.mark.xfail(reason="Missing IMAP specific global schema")
+
+
 def test_global_attribute():
-    cdf_manager = CdfAttributeManager(Path(__file__).parent.parent / "config")
+    """
+    Test function that covers:
+        load_global_attributes
+    """
 
+    # Initialize CdfAttributeManager object which loads in default info
+    cdf_manager = CdfAttributeManager(Path(__file__).parent.parent / "config")
     cdf_manager.load_global_attributes("imap_default_global_cdf_attrs.yaml")
 
-    # Expected failure: file_naming_convention is not in schema.
-    # for attr in cdf_manager.global_attributes.keys():
-    #     assert attr in cdf_manager.global_attribute_schema.keys()
-
+    # Testing information has been loaded in
+    assert cdf_manager.global_attributes["Project"] == "STP>Solar-Terrestrial Physics"
+    assert (
+        cdf_manager.global_attributes["Source_name"]
+        == "IMAP>Interstellar Mapping and Acceleration Probe"
+    )
+    assert (
+        cdf_manager.global_attributes["Discipline"]
+        == "Solar Physics>Heliospheric Physics"
+    )
     assert (
         cdf_manager.global_attributes["Mission_group"]
         == "IMAP>Interstellar Mapping and Acceleration Probe"
     )
+    assert cdf_manager.global_attributes["PI_name"] == "Dr. David J. McComas"
+    assert (
+        cdf_manager.global_attributes["PI_affiliation"]
+        == "Princeton Plasma Physics Laboratory, 100 Stellarator Road, "
+        "Princeton, NJ 08540"
+    )
+    assert (
+        cdf_manager.global_attributes["File_naming_convention"]
+        == "source_descriptor_datatype_yyyyMMdd_vNNN"
+    )
+    # The following test should fail because "DOI" is not an attribute in
+    #   imap_default_global_cdf_attrs.yaml
+    with pytest.raises(KeyError):
+        assert cdf_manager.global_attributes["DOI"] == "test"
 
-    # Load additional global attributes
-    cdf_manager.load_global_attributes("imap_mag_global_cdf_attrs.yaml")
+    # Load in different data
+    cdf_manager.source_dir = Path(__file__).parent.parent / "tests"
+    cdf_manager.load_global_attributes("imap_default_global_test_cdf_attrs.yaml")
 
+    # Testing attributes carried over
     assert (
-        cdf_manager.global_attributes["Mission_group"]
+        cdf_manager.global_attributes["File_naming_convention"]
+        == "source_descriptor_datatype_yyyyMMdd_vNNN"
+    )
+    assert (
+        cdf_manager.global_attributes["Discipline"]
+        == "Solar Physics>Heliospheric Physics"
+    )
+
+    # Testing attributes newly loaded
+    assert cdf_manager.global_attributes["Project"] == "STP>Solar-Terrestrial Physics"
+    assert (
+        cdf_manager.global_attributes["Source_name"]
         == "IMAP>Interstellar Mapping and Acceleration Probe"
     )
+    assert cdf_manager.global_attributes["Mission_group"] == "Dysfunctional Cats"
```

Comment:
Ahh ok I see what you're testing here. I like that you're testing that the attributes are overwritten, I would include two test files (test_1 and test_2) and not only test that the attributes are overwritten (like you are here) and that you can add new attributes (like with DOI) but also that you are not required to overwrite attributes (so, for example, remove "Data_version" from your second attribute file and assert that it's the same as in the first attribute file.)